### PR TITLE
hpo selects the best regression model

### PR DESCRIPTION
**Pull request by tdhd**
*Description of changes:*

HPO for a regression model selected the worst performing HP configuration according to MSE.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/test_simple_imputer.py [hunk 1]
```diff
@@ -808,3 +808,32 @@ def assert_val(col, value):
 
     assert_val('global:final_fc_hidden_units', [1])
     assert_val('global:learning_rate', 0.1)
+
+
+def test_hpo_numeric_best_pick(test_dir, data_frame):
+    feature_col, label_col = "feature", "label"
+
+    df = data_frame(feature_col=feature_col,
```
Comment (felixbiessmann) [inline test/test_simple_imputer.py]: there is some data generator for this test which generates maybe slightly more complicated data for a more realistic test:
https://github.com/awslabs/datawig/blob/master/datawig/utils.py#L311

File: test/test_simple_imputer.py [hunk 2]
```diff
@@ -808,3 +808,32 @@ def assert_val(col, value):
...
+
+    results = imputer.hpo.results
+
+    max_tokens_of_encoder = imputer.imputer.data_encoders[0].vectorizer.max_features
+
+    # model with minimal MSE
+    best_hpo_run = imputer.hpo.results['mse'].astype('float').idxmin()
+    loaded_hpo_run = results.loc[results[feature_col+':max_tokens'] == max_tokens_of_encoder].index[0]
+
+    assert best_hpo_run == loaded_hpo_run
```
Comment (felixbiessmann) [inline test/test_simple_imputer.py]: is checking the run index safe or should we test the hps instead?

File: datawig/simple_imputer.py [hunk 3]
```diff
@@ -563,15 +563,16 @@ def load_hpo_model(self, hpo_name: int = None):
...
 
         if self.hpo.results is None:
-            logger.warn('No hpo run available. Run hpo calling SimpleImputer.fit_hpo().')
+            logger.warn('No hpo run available. Run hpo by calling SimpleImputer.fit_hpo().')
             return
 
         if hpo_name is None:
             if self.output_type == 'numeric':
-                hpo_name = self.hpo.results['mse'].astype(float).idxmax()
+                hpo_name = self.hpo.results['mse'].astype(float).idxmin()
```
Comment (felixbiessmann) [inline datawig/simple_imputer.py]: great, thanks! did we also fix the metrics now so that the right metric name is logged when optimizing?

File: test/test_simple_imputer.py (hunk 1 above)
Comment (tdhd) [reply test/test_simple_imputer.py]: oh great, I didn't know that, I'll update it to use that.

File: test/test_simple_imputer.py (hunk 2 above)
Comment (tdhd) [reply test/test_simple_imputer.py]: the run index is the identifier for the HPO run, it is also used for loading the best HPO model in `SimpleImputer`.

File: datawig/simple_imputer.py (hunk 3 above)
Comment (tdhd) [reply datawig/simple_imputer.py]: I didn't change how metrics are logged.

File: test/test_simple_imputer.py (hunk 1 above)
Comment (tdhd) [reply test/test_simple_imputer.py]: Though this test case is merely about asserting the ordering of the models and not necessarily based on which input data the model depends on.
